Source/SwiftLintCore/Models/Baseline.swift

```diff
@@ -0,0 +1,169 @@
+        )
+
+        var filteredViolations: Set<BaselineViolation> = []
+
+        for (ruleIdentifier, ruleViolations) in violationsByRuleIdentifier {
+            guard
+                let baselineViolations = baselineViolationsByRuleIdentifier[ruleIdentifier],
+                    baselineViolations.isNotEmpty else {
+                filteredViolations.formUnion(ruleViolations)
+                continue
+            }
+
+            let groupedRuleViolations = Dictionary(grouping: ruleViolations) { $0.key }
+            let groupedBaselineViolations = Dictionary(grouping: baselineViolations) { $0.key }
+
+            for (key, ruleViolations) in groupedRuleViolations {
+                guard let baselineViolations = groupedBaselineViolations[key] else {
+                    filteredViolations.formUnion(ruleViolations)
+                    continue
+                }
+                if ruleViolations.count > baselineViolations.count {
+                    filteredViolations.formUnion(ruleViolations)
+                }
+            }
+        }
+
+        let violationsWithAbsolutePaths = Set(filteredViolations.violationsWithAbsolutePaths)
+        return violations.filter { violationsWithAbsolutePaths.contains($0) }
+    }
+}
+
+private extension Sequence where Element == StyleViolation {
+    var baselineViolations: [BaselineViolation] {
+        var lines: [String: [String]] = [:]
+        var baselineViolations: [BaselineViolation] = []
+        for violation in self {
+            guard let absolutePath = violation.location.file,
+                  let lineNumber = violation.location.line != nil ? violation.location.line! - 1 : nil,
+                  lineNumber > 0 else {
+                baselineViolations.append(BaselineViolation(violation: violation, text: ""))
+                continue
+            }
+            if let fileLines = lines[absolutePath] {
+                let text = (!fileLines.isEmpty && lineNumber < fileLines.count ) ? fileLines[lineNumber] : ""
+                baselineViolations.append(BaselineViolation(violation: violation, text: text))
+                continue
+            }
+            let text: String
+            if let fileLines = SwiftLintFile(path: absolutePath)?.lines.map({ $0.content }),
+               lineNumber < fileLines.count {
+                text = fileLines[lineNumber]
+                lines[absolutePath] = fileLines
+            } else {
+                text = ""
+            }
+            baselineViolations.append(BaselineViolation(violation: violation, text: text))
+        }
+        return baselineViolations
+    }
+}
+
+private extension Sequence where Element == BaselineViolation {
+    var violationsWithAbsolutePaths: [StyleViolation] {
+        map {
+            let location = $0.violation.location
+            let absolutePath: String?
+            if let relativePath = location.file {
+                absolutePath = FileManager.default.currentDirectoryPath + "/" + relativePath
```

Comment:
So that is true - we also suffer from that problem in some of the reporters (see screenshot below). I think `relativeFile` is going to be important in a group setting, where the baseline is likely generated on a different machine from the developers.

I guess we could have an option to use absolutePaths instead (Periphery, in general, seems to use absolute paths, but has an option to use relative paths) if you think we need it.

<img width="454" alt="Screen Shot 2024-04-13 at 19 58 27" src="https://github.com/realm/SwiftLint/assets/435558/b7ad75c8-1a35-4fb8-a588-d8f95e61346d">